lib/fluent/plugin/output.rb

```diff
@@ -825,6 +822,16 @@ def metadata(tag, time, record)
         end
       end
 
+      def calculate_timekey(time)
+        time_int = time.to_i
+        if @buffer_config.timekey_use_utc
```

Comment:
accessing object member is slower than accessing instance variable.
`@timekey_use_utc = @buffer_config.timekey_use_utc` in `configure` is better.